### PR TITLE
Added local config file overwrites

### PR DESCRIPTION
**Pull request by pchalupa**
# Proposed Changes
  - Rename common config
  - Added config local overwrites

## Summary
Added possibility to overwrite config by adding local in his name. Example: `common.local.ini`, will resolve to load this file instead of base `config.ini` file.

### Checklist
- [ ] Unit tests for the changes have been added
- [x] Static analysis exits without any error
- [x] CS Fix exits without any error

### CODE REVIEW

**Comment by oliverstasa** [comment]
where is the config called?
the solution i can't grasp is that when you create multiple configs, how can the api tell which one to pick? or there's unwritten rule to keep just one config file in the directory - that would solve it

**Comment by pchalupa** [comment]
> where is the config called?
> the solution i can't grasp is that when you create multiple configs, how can the api tell which one to pick? or there's unwritten rule to keep just one config file in the directory - that would solve it

In a config directory are these files: `common.ini, common.local.ini, api.ini`. The way how it works is pick file by his name, like common. If there is local present use that file instead. So every local config file is related only with current instance.
Mean you can have multiple configs in config directory.

File: src/Lib/Configuration/ConfigurationFactory.php [hunk 1]
```diff
@@ -9,20 +11,22 @@ final class ConfigurationFactory
...
      *
      * @param string $fileName
      * @return Configuration
      */
-    public static function fromFileName(string $fileName):Configuration
+    public static function fromFileName(string $fileName): Configuration
     {
-        if (empty($fileName) || !($path = realpath(self::DIRECTORY . $fileName . '.ini'))) {
-            throw new Exception('Config file ' . $fileName . ' not exists!');
+        $configs = glob(self::CONFIG_DIRECTORY . $fileName . '*' . 'ini');
```
Comment (oliverstasa) [inline src/Lib/Configuration/ConfigurationFactory.php:24]: missing a dot?
`$fileName . '*' . '.ini');`

non the less i have no idea why there sould be '*', when you pass the file name somewhere else

**Review by oliverstasa — APPROVED** [review]

File: src/Lib/Configuration/ConfigurationFactory.php (hunk 1 above)
Comment (pchalupa) [reply src/Lib/Configuration/ConfigurationFactory.php]: True, fixed!

**Pull request merged**